source/common/upstream/cds_api_impl.cc

```diff
@@ -51,12 +47,44 @@ void CdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::stri
   for (const auto& cluster : resources) {
     MessageUtil::validate(cluster);
   }
-  // We need to keep track of which clusters we might need to remove.
+
```

Comment:
Shouldn't we move the above duplication and validation logic to the delta `onConfigUpdate`? We want to do this in both cases.

Reply:
Done.